unit-test/fm_child_tests.c

```diff
@@ -3286,6 +3262,44 @@ void Test_FM_ChildDirListFileLoop_BytesWrittenNotEqualWriteLength(void)
     UtAssert_INT32_EQ(count_fileentries, 1);
 }
 
+void Test_FM_ChildDirListFileLoop_BytesWrittenNotEqualWriteLengthInLoop(void)
+{
+    // Arrange
+    osal_id_t   dirid                 = FM_UT_OBJID_1;
+    osal_id_t   fileid                = FM_UT_OBJID_2;
+    os_dirent_t dummy_direntry        = {.FileName = "dummy_directory_nam"};
+    const char *dummy_directory       = "dummy_directory";
+    const char *dummy_filename        = "dummy_filename";
+    const char *dummy_dirwithstep     = "dummy_dirwithstep";
+    uint8       dummy_getsizetimemode = 1;
+
+    FM_GlobalData.ChildCmdErrCounter  = 1;
+    FM_GlobalData.ChildCmdCounter     = 1;
+    FM_GlobalData.ChildCmdWarnCounter = 1;
```

Comment:
If you leave these as zero you can remove the local count_* variables below and assert against just the expected value (0 or 1 as needed) vs an increment.

Reply:
Thank you. The FM_GlobalData counters and asserts have been updated in Test_FM_ChildDirListFileLoop_BytesWrittenNotEqualWriteLengthInLoop().